### PR TITLE
Validate timestamp in Windows

### PR DESCRIPTION
**Pull request by msftrubengu**
Add a call to WinVerifyTrust to validate Timestamp on msix package.

### CODE REVIEW

**Review by yao-msft — APPROVED** [review]

File: src/msix/PAL/Signature/Win32/SignatureValidator.cpp [hunk 1]
```diff
@@ -127,8 +127,8 @@ namespace MSIX
         ThrowErrorIf(Error::SignatureInvalid, (signingCertContext.get() == NULL), "failed to get signing cert context.");
 
         // Get the signing certificate chain context.  Do not connect online for URL 
-        // retrievals.  Note that this check does not respect the lifetime signing 
-        // EKU on the signing certificate.  
+        // retrievals. If CertVerifyCertificateChainPolicy fail to validate the certificates
```
Comment (ranm-msft) [inline src/msix/PAL/Signature/Win32/SignatureValidator.cpp:130]: >fail [](start = 59, length = 4)

fails

File: src/msix/PAL/Signature/Win32/SignatureValidator.cpp [hunk 2]
```diff
@@ -127,8 +127,8 @@ namespace MSIX
         ThrowErrorIf(Error::SignatureInvalid, (signingCertContext.get() == NULL), "failed to get signing cert context.");
 
         // Get the signing certificate chain context.  Do not connect online for URL 
-        // retrievals.  Note that this check does not respect the lifetime signing 
-        // EKU on the signing certificate.  
+        // retrievals. If CertVerifyCertificateChainPolicy fail to validate the certificates
+        //  we call WinVerifyTrust, if return ERROR_SUCCESS or "0" the signature is valid.
```
Comment (ranm-msft) [inline src/msix/PAL/Signature/Win32/SignatureValidator.cpp:131]: >WinVerifyTrust [](start = 20, length = 14)

which also checks the timestamp (if a package is timestamped whil the cert is valid, we consider the package current)

File: src/msix/PAL/Signature/Win32/SignatureValidator.cpp [hunk 3]
```diff
@@ -598,19 +631,30 @@ namespace MSIX
         );
 
         origin = MSIX::SignatureOrigin::Unknown;
-        if (IsStoreOrigin(p7s, p7sSize)) { origin = MSIX::SignatureOrigin::Store; }
-        else if (IsAuthenticodeOrigin(p7s, p7sSize)) { origin = MSIX::SignatureOrigin::LOB; }
+        if (IsStoreOrigin(p7s, p7sSize))
+        {
+            origin = MSIX::SignatureOrigin::Store;
+        }
+        else if (IsAuthenticodeOrigin(p7s, p7sSize))
```
Comment (ranm-msft) [inline src/msix/PAL/Signature/Win32/SignatureValidator.cpp:638]: >else [](start = 8, length = 4)

comment:
// certChainContext is called to validate the cert signature

File: src/msix/PAL/Signature/Win32/SignatureValidator.cpp [hunk 4]
```diff
@@ -598,19 +631,30 @@ namespace MSIX
...
-        else if (IsAuthenticodeOrigin(p7s, p7sSize)) { origin = MSIX::SignatureOrigin::LOB; }
+        if (IsStoreOrigin(p7s, p7sSize))
+        {
+            origin = MSIX::SignatureOrigin::Store;
+        }
+        else if (IsAuthenticodeOrigin(p7s, p7sSize))
+        {
+            origin = MSIX::SignatureOrigin::LOB;
+        }
+        // WinVerifyTrust is called to validate the certificate signature.
```
Comment (ranm-msft) [inline src/msix/PAL/Signature/Win32/SignatureValidator.cpp:642]: >WinVerifyTrust [](start = 11, length = 14)

is called to validate timestamp

**Review by ranm-msft — APPROVED** [review]
:shipit:

**Pull request merged**